### ISSUE TITLE
Add activitly relationships to the model

### ISSUE THREAD

**Issue by 23bartman** [owaspsamm/samm-suite#62]
**Description:**
In the model, many activities are implicitly related to each other. Sometimes an activity is a prerequisite for another activity, sometimes activities reinforce one another, and so on. Since these relationships are not represented as such in the model, it is often not clear to users if and how activities are related.

Relationships between activities are useful to understand for the community.  We can extend the model to represent these relationships more explicitly.

**Acceptance criteria:**
- Define the different types of relationships that we want to support
- Analyse the relationships between activities
- Add the relationship information to the YAML files 

**Are there any known dependencies between this issue and any other issues?**

**Are there any outstanding questions?**
- Do we need an update to the YAML template to support this (currently there is a relationship items, but that might not be sufficient)

**Comment by aramhovsepyan** [comment]
How far shall we go in terms of relationship strength? Some activities are strongly linked (e.g., TM and Arch Mitigation). Others could be considered linked, but the strength is much weaker.
Also, is this a model update or an "internal mapping from SAMM to SAMM" like we did with other frameworks?

**Comment by aganariman** [comment]
Isn't this problem already being solved by the OpenCRE project on a larger scale? "Which is related to CREs" section shows related other SAMM activities.

**Comment by 23bartman** [comment]
Good point @aganariman. We can use this as a basis/starting point for this issue. However, it is quite cumbursome to look up relations like this, so an easier way to look this up would be preferred. Also, it would be best if we drive these relationships from within the model. And we might want to make relationships more specific (e.g., "prerequisite for" vs. "related" vs. "..."). So we definitely need to look into this one further IMO.

**Comment by aramhovsepyan** [comment]
Indeed, we would like to go beyond "crossover" mappings by adding more semantics to it.

**Comment by aramhovsepyan** [comment]
Creating this spreadsheet to start working on hard dependencies
https://docs.google.com/spreadsheets/d/1fI6XNzgEv7H975SVEWCLassDQdXhHPJbi1QkWx_6VbY/edit?gid=0#gid=0

**Comment by Pat-Duarte** [comment]
Review in this format. Core Team review. Then, community review.
Later, define whether we're adding this to the YAML files.

**Comment by dkefer** [comment]
Hi @aramhovsepyan , I've left some inline comments in the linked spreadsheet. I think there is quite some potential for discussion / controversies related to "where exactly is the trigger point for when we start considering an activity A to be a dependency to activity B". I personally see the following possibilities: 
- Try to limit the number of dependencies as much as possible for clear and non-negotiable cases (e.g. "you need security requirements to do requirements-driven testing", or "you need security trainings to be able to improve these"). Not sure how much should we imply dependencies just because there is one sentence or one quality criteria seeming a bit artificial to me, e.g. D-SA-1-B depending on G-SM-1-A. Maybe we could use the model update to rather remove such cases?
- Potentially introduce more types of relations or a scale of dependency - would complicate the project and not sure how many users would actually benefit from that.
- Don't put the dependencies into the data model, rather describe common dependencies in a separate document. Here I'd see the benefit to be able to talk about relationships among various entities on different levels (e.g. verification activities vs defect management and CI/CD, dependency of all activities on governance function....)

**Comment by 23bartman** [comment]
I added my view on this as well in the sheet. I do think that we need a dedicated discussion session to take some decisions on some topics. To be scheduled.

**Comment by Pat-Duarte** [comment]
Discussed during the May core team call.
We need more time. Schedule a meeting (June 18)

**Comment by Pat-Duarte** [comment]
Focus on the most evident relationships and decide if we'll include them in the model.

**Comment by Pat-Duarte** [comment]
There's agreement on the relationships. Next step is to include them in the model.

**Comment by Pat-Duarte** [comment]
Create issues for some things that came up during meetings - @23bartman 
Dependencies are implemented in a feature branch by @23bartman: Verify

**Comment by johnellingsworth** [comment]
fixed by https://github.com/owaspsamm/core/pull/186
